src/main/java/org/jitsi/videobridge/LipSyncHack.java

```diff
@@ -497,12 +498,33 @@ public void run()
         }
 
         /**
-         * Schedules this instance for execution at a fixed rate.
+         * {@inheritDoc}
          */
-        public void schedule()
+        @Override
+        public long getTimeUntilNextRun()
         {
-            this.scheduledFuture = scheduler.scheduleAtFixedRate(
-                this, WAIT_MS , KEY_FRAME_RATE_MS, TimeUnit.MILLISECONDS);
+            synchronized (injectState)
+            {
+                if (!injectState.active
+                    || injectState.numOfKeyframesSent >= MAX_KEY_FRAMES)
+                {
+                    return -1L;
```

Comment:
It is not clear from the documentation of RecurringRunnable what the semantics of returning -1 are, but from a quick look at the code it looks like this would make RecurringRunnableExecutor run the task perpetually.